src/v0/destinations/intercom_v2/transform.js

```diff
@@ -48,6 +49,10 @@ const transformTrackPayload = (event) => {
   if (!(payload.user_id || payload.email)) {
     throw new InstrumentationError('Either email or userId is required for Track call');
   }
+  if (!payload.id) {
+    const integrationsObj = getIntegrationsObj(message, 'INTERCOM');
+    payload.id = integrationsObj?.id;
+  }
```

Comment:
If the customer has provided intercom id through integrations object, we can use that to make the track call. We can skip the check for email, userId.